File: applications/luci-app-zerotier/htdocs/luci-static/resources/view/network/zerotier.js
'use strict';
'require view';
'require form';

return view.extend({
  render: function() {
    var m, s, o;

    m = new form.Map('zerotier', _('ZeroTier'));

    s = m.section(form.TypedSection, 'zerotier');
    s.addremove = true;

    o = s.option(form.Flag, 'enabled', _('Enabled'));
    o.default = '0';
    o.rmempty = false;

    s.option(form.Value, 'port', _('Port'));

    s.option(form.DynamicList, 'join', _('Networks'));

    return m.render();
  },
});

File: applications/luci-app-zerotier/luasrc/model/cbi/zerotier.lua
-- Copyright 2021 Jonathan Kolb <kolbyjack@gmail.com>
-- Licensed to the public under the Apache License 2.0.

local packageName = "zerotier"
local sys = require "luci.sys"

local m = Map("zerotier", translate("ZeroTier Settings"))

s = m:section(TypedSection, "zerotier")
s.addremove = true

o = s:option(Flag, "enabled", translate("Enabled"))
o.default = "0"
o.rmempty = false

s:option(Value, "port", translate("Port"))

s:option(DynamicList, "join", translate("Networks"))

return m
